schema/me/conversation.js

```diff
@@ -1,11 +1,131 @@
+import date from "../fields/date"
 import impulse from "../../lib/loaders/impulse"
 import gravity from "../../lib/loaders/gravity"
+import { get } from "lodash"
+import { GraphQLBoolean, GraphQLList, GraphQLObjectType, GraphQLString, GraphQLNonNull, GraphQLEnumType } from "graphql"
+import { ArtworkType } from "../artwork"
+const { IMPULSE_APPLICATION_ID } = process.env
 
-import { GraphQLString, GraphQLNonNull } from "graphql"
+export const BuyerOutcomeTypes = new GraphQLEnumType({
+  name: "BuyerOutcomeTypes",
+  values: {
+    PURCHASED: {
+      value: "purchased",
+    },
+    STILL_CONSIDERING: {
+      value: "still_considering",
+    },
+    HIGH_PRICE: {
+      value: "high_price",
+    },
+    LOST_INTEREST: {
+      value: "lost_interest",
+    },
+    WORK_UNAVAILABLE: {
+      value: "work_unavailable",
+    },
+    OTHER: {
+      value: "other",
+    },
+    BLANK: {
+      value: "",
+    },
+  },
+})
 
-import { ConversationType } from "./conversations"
+export const MessageType = new GraphQLObjectType({
+  name: "MessageType",
+  description: "A message in a conversation.",
+  fields: {
+    id: {
+      description: "Impulse id.",
+      type: GraphQLString,
+    },
+    from_email_address: {
+      description: "Email address of sender.",
+      type: GraphQLString,
+    },
+    snippet: {
+      description: "A snippet of the full message",
+      type: GraphQLString,
+    },
+  },
+})
 
-const { IMPULSE_APPLICATION_ID } = process.env
+export const ConversationFields = {
+  id: {
+    description: "Impulse id.",
+    type: GraphQLString,
+  },
+  inquiry_id: {
+    description: "Gravity inquiry id.",
+    type: GraphQLString,
+  },
+  from_id: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  from_type: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  from_name: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  from_email: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  to_id: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  to_type: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  to_name: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
```

Comment:
so the conversation as it is in Impulse, can only be initiated by one User or Partner, the combination of `from_id` and `from_type` defines that. As for `to` it's kind of the same deal it can be to a Partner (in which case we get list of all the contacts in that partner who receive inquiry email) or can be to a User. In current Impulse data all of the conversations are started by a user (from is always a user) and sent to a Partner (to is always a partner).

In Radiation messages we track who was involved in one specific message of a conversation, as an example in CMS where we can reply to a message:
<img width="664" alt="screen shot 2017-06-14 at 8 10 06 am" src="https://user-images.githubusercontent.com/1230819/27131533-0269a64e-50d9-11e7-8424-edf50b36bbb5.png">

That `to` list is populated using Radiation (remember Radiation is the one who knows about masked emails and Impulse knows about unmasked).

In this flow, if I reply to that message, we will add one record to Radiation that has 4 _deliveries_ to all people in "to" list and we will have 1 new message record in Impulse where `from_email_address` is "from" address of that message (in this case ashkan.nasseri@gmail.com) and snippet is first 500 characters.